### ISSUE TITLE
re-run failed tests triggers on failed coverage

### ISSUE THREAD

**Issue by chrismeyersfsu** [ansible/awx-plugins#21]
https://github.com/ansible/awx-plugins/actions/runs/10597353901/job/29367631346?pr=17

`FAIL Required test coverage of 100% not reached. Total coverage: 40.78%`

It looks like the re-run failed tests with higher verbosity triggers re-running ALL tests when the coverage check fails. In the case above all the tests passed both runs.

**Comment by webknjaz** [comment]
So… Re-runs are made to always fail so that they don't hide flakiness. There's a `|| exit 1` in the command. That is intentional.

The only problematic bit I noticed are tracebacks coming from `commands_post`.

Any other action item I'm missing?

**Comment by chrismeyersfsu** [comment]
I think I'm wrong about coverage failure being the reason.

I was trying to point out that all tests pass (first screenshot)
![image](https://github.com/user-attachments/assets/ba33a155-4cab-437a-b1ea-e0d5da80b20e)

Then the re-run seems to run ALL the tests. I would expect it to only re-run the tests that failed above (of which there are none) (second screenshot)
![image](https://github.com/user-attachments/assets/1c778b55-4b7e-46f9-8b45-9e5d6ecef7d9)


There is also an array index error but I think that may be related to coverage.

**Comment by webknjaz** [comment]
No, the index error is related to the last verbose tox call w/o coverage collection. I only tested the first one because there were no failures in the skeleton. So reruns weren't triggered. I fixed that already.

The reruns are generic and trigger on any errors in previous GHA steps.

The pytest rerun is invoked with `--lf` to pick up the failed tests but since there's none, all are executed again. I think you're right that coverage likely causes the failure of that first testing invocation.

**Comment by webknjaz** [comment]
@chrismeyersfsu do you have a preference / any ideas on how you'd like this to work?
